pkg/controller/storagecluster/cephobjectstores.go

```diff
@@ -33,10 +33,18 @@ func (r *ReconcileStorageCluster) ensureCephObjectStores(instance *ocsv1.Storage
 	if err != nil {
 		return err
 	}
+
+	r.createCephObjectStores(cephObjectStores, instance, reqLogger)
```

Comment:
This returns an error, but it seems you're ignoring it. You should check for the error and only set the Status boolean true if err == nil.